pennylane/qchem/factorization.py

```diff
@@ -226,28 +362,145 @@ def _double_factorization_cholesky(two, tol_factor=1.0e-10, shape=None, interfac
         eigenvalues of the generated factors, and truncated eigenvectors of the generated factors
     """
     n2 = shape[0] * shape[1]
-    cholesky_vecs = qml.math.zeros((n2, n2 + 1), like=interface)
+    if num_factors is None:
+        num_factors = n2
+
+    cholesky_vecs = qml.math.zeros((n2, num_factors), like=interface)
     cholesky_diag = qml.math.array(qml.math.diagonal(two).real, like=interface)
 
-    for idx in range(n2 + 1):
+    for idx in range(num_factors):
```

Comment:
Since we are using the old code, here num_factors is equal to n2, whereas earlier we were using n2+1. Was their a mistake in the earlier code or has the logic changed somewhat?

Reply:
Not a mistake but the choice of n2+1 was rather arbitrary. I made it n2 to keep it consistent with the eigendecomposition case.